doc/source/operations/nova-compute-ironic.rst

```diff
@@ -0,0 +1,211 @@
+===================
+Nova Compute Ironic
+===================
+
+This section describes the deployment of the OpenStack Nova Compute
+Ironic service. The Nova Compute Ironic service is used to integrate
+OpenStack Ironic into Nova as a 'hypervisor' driver. The end users of Nova
+can then deploy and manage baremetal hardware, in a similar way to VMs.
+
+High Availability (HA)
+======================
+
+The OpenStack Nova Compute service is designed to be installed once on every
+hypervisor in an OpenStack deployment. In this configuration, it makes little
+sense to run additional service instances. Even if you wanted to, it's not
+supported by design. This pattern breaks down with the Ironic baremetal
+service, which must run on the OpenStack control plane. It is not feasible
+to have a 1:1 mapping of Nova Compute Ironic services to baremetal nodes.
+
+The obvious HA solution is to run multiple instances of Nova Compute Ironic
+on the control plane, so that if one fails, the others can take over. However,
+due to assumptions long baked into the Nova source code, this is not trivial.
+The HA feature provided by the Nova Compute Ironic service has proven to be
+unstable, and the direction upstream is to switch to an active/passive
+solution [1].
+
+However, challenges still exist with the active/passive solution. Since the
+Nova Compute Ironic HA feature is 'always on', one must ensure that only a
+single instance (per Ironic conductor group) is ever running. It is not
+possible to simply put multiple service instances behind HAProxy and use the
+active/passive mode.
+
+Such problems are commonly solved with a technology such as Pacemaker, or in
+the modern world, with a container orchestration engine such as Kubernetes.
+Kolla Ansible provides neither, because in general it doesn't need to. Its
+goal is simplicity.
+
+The interim solution is to therefore run a single Nova Compute Ironic
+service. If the service goes down, remedial action must be taken before
+Ironic nodes can be managed. In many environments the loss of the Ironic
+API for short periods is acceptable, providing that it can be easily
+resurrected. The purpose of this document is to faciliate that.
+
+TODO: Add caveats about new sharding mode (not covered here).
+
+Optimal configuration of Nova Compute Ironic
+============================================
+
+Determine the current configuration for the site. How many Nova Compute
+Ironic instances are running on the control plane?
+
+.. code-block:: console
+
+  $ openstack compute service list
+
+Typically you will see either three or one. By default the host will
+marked with a postfix, eg. ``controller1-ironic``. If you find more than
+one, you will need to remove some instances. You must complete the
+following section.
+
+Moving from multiple Nova Compute Instances to a single instance
+----------------------------------------------------------------
+
+1. Decide where the single instance should run. Typically, this will be
```

Comment:
What would be a good method of deciding where the single instance should run?

Reply:
I've added some words - generally any controller.